File: kardex/src/main/java/com/example/jumunoz/kardex/services/ProductService.java
package com.example.jumunoz.kardex.services;

import java.util.List;

import com.example.jumunoz.kardex.model.Product;

public interface ProductService {

    public List<Product> getAllProducts();

    public Product getProductById(int id);

    public Product addNewProduct(Product product);

    public Product updateProduct(Product product);

}

File: kardexSystem-angular/src/app/service/product.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Product } from '../model/product';

@Injectable({
    providedIn: 'root'
})

export class ProductService {

    private baseUrl = 'http://localhost:8888/kardex/product';

    constructor(private http: HttpClient) { }

    getProducts(): Observable<any> {
        return this.http.get(this.baseUrl);
    }

    addNewProduct(product: Product): Observable<any> {
        return this.http.post(this.baseUrl + '/add', product);
    }

    getProductById(id: number): Observable<any> {
        return this.http.get(this.baseUrl + '/' + id);
    }

    updateStock(product: Product): Observable<any> {
        return this.http.put(this.baseUrl + '/update/' + product.id, product);
    }
}
